### PR TITLE
2548 reserve rates api

### PR DESCRIPTION
**Pull request by cyberliem**

### CODE REVIEW

File: accounting/reserve-rate/http/http.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+func (sv *Server) reserveRates(c *gin.Context) {
+	var (
+		query  httputil.TimeRangeQuery
+		logger = sv.sugar.With("func", "accouting/reserve-rate/http/Server.reserveRates")
+	)
+
+	if err := c.ShouldBindQuery(&query); err != nil {
+		c.JSON(
+			http.StatusBadRequest,
+			gin.H{"error": err.Error()},
```
Comment (favadi) [inline accounting/reserve-rate/http/http.go]: should be replaced with httputil.ResponseFailure, same for other error request in this file.

File: tradelogs/storage/influx.go [hunk 2]
```diff
@@ -262,14 +262,14 @@ func (is *InfluxStorage) getWalletFeeAmount(log common.TradeLog) (float64, float
...
-			}
-		} else {
+		switch {
+		case walletFee.ReserveAddress == log.SrcReserveAddress && !srcAmountSet:
+			srcAmount = amount
+			// to prevent setting SrcReserveAddress twice when SrcReserveAddress =DstReserveAddress
+			srcAmountSet = true
+		case walletFee.ReserveAddress == log.DstReserveAddress:
+			dstAmount = amount
+		default:
```
Comment (favadi) [inline tradelogs/storage/influx.go]: this looks unrelated to the task.

File: accounting/cmd/accounting-reserve-rate-api/main.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import (
+	"log"
+	"os"
+
+	"github.com/KyberNetwork/reserve-stats/accounting/common"
+	"github.com/KyberNetwork/reserve-stats/accounting/reserve-rate/http"
+	rrpostgres "github.com/KyberNetwork/reserve-stats/accounting/reserve-rate/storage/postgres"
+	libapp "github.com/KyberNetwork/reserve-stats/lib/app"
+	"github.com/KyberNetwork/reserve-stats/lib/httputil"
+	"github.com/urfave/cli"
```
[the remark this answers is not in the document]
Comment (cyberliem) [reply accounting/cmd/accounting-reserve-rate-api/main.go]: imported

File: accounting/cmd/accounting-reserve-rate-api/main.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		return err
+	}
+	defer func(err *error) {
+		cErr := ratesStorage.Close()
+		if err == nil {
+			*err = cErr
+		} else {
+			sugar.Error("DB closing failed", "error", cErr)
+		}
+	}(&err)
```
[the remark this answers is not in the document]
Comment (cyberliem) [reply accounting/cmd/accounting-reserve-rate-api/main.go]: changed

File: accounting/reserve-rate/http/http.go [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import (
+	"net/http"
+	"time"
+
+	"github.com/KyberNetwork/reserve-stats/accounting/reserve-rate/storage"
+	"github.com/KyberNetwork/reserve-stats/lib/httputil"
+	_ "github.com/KyberNetwork/reserve-stats/lib/httputil/validators" // import custom validator functions
+
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
```
[the remark this answers is not in the document]
Comment (cyberliem) [reply accounting/reserve-rate/http/http.go]: imported

File: accounting/reserve-rate/http/http.go (hunk 1 above)
Comment (cyberliem) [reply accounting/reserve-rate/http/http.go]: replaced.

File: tradelogs/storage/influx.go (hunk 2 above)
Comment (cyberliem) [reply tradelogs/storage/influx.go]: reverted
